teamscale-client/src/main/java/com/teamscale/client/TeamscaleServer.java

```diff
@@ -69,4 +74,16 @@ public String toString() {
 		}
 		return "Teamscale " + url + " as user " + userName + " for " + project + " to " + partition + " at " + at;
 	}
+
+	/** Creates a copy of the {@link TeamscaleServer} configuration, but with the given project and commit set. */
+	public TeamscaleServer withProject(String teamscaleProject, CommitDescriptor commitDescriptor) {
+		TeamscaleServer tsApplication = new TeamscaleServer();
```

Comment:
why is this called tsApplication? Seems to be a server with a project and a commit.